arm/Microsoft.DataFactory/factories/.bicep/nested_rbac.bicep

```diff
@@ -20,6 +20,7 @@ var builtInRoleNames = {
 }
 
 resource dataFactory 'Microsoft.DataFactory/factories@2018-06-01' = {
```

Comment:
Rookie mistake ;) 

In bicep we use the concept of an `existing` resource to allow us to use it as a reference in files without actually deploying it.

What your solution should look like is this:
```bicep
resource dataFactory 'Microsoft.DataFactory/factories@2018-06-01' existing = {
  name: resourceName
}
```

If you'd convert this bad boy to JSON you'll notice that ARM converts this to a simple resource reference - i.e. it is not a resource deployment

Reply:
Thank you for calling this out! Done for the Data Factory. We may have to go through modules and see if this is occurring on others :)